### DISCUSSION TITLE
the colors of this picture will change a bit. Is there any way to keep the original colors? Thank you!

### DISCUSSION THREAD

**Discussion by lichdb** [ImageMagick/ImageMagick#7774] [Help]
### ImageMagick version

7.1.1-37 Q16 x64

### Operating system, version and so on

windows 10 

### Description

The picture raw1.dng was converted into raw1.jpg through ImageMagick. There is a slight color change compared to the original picture. Is there any way to keep the original colors? Thank you!

![image](https://github.com/user-attachments/assets/95f44580-3043-4f23-a9b2-6db9b3264347)

**Comment by snibgo** [comment]
> There is a slight color change compared to the original picture.

What "original picture"? The DNG file contains a raw image that does not contain colours. To make a colour image from that, some process must demosaic the image and transform it with a camera matrix. The process may also do other operations. This is what IM does when it reads a DNG file. See also the `dng:` defines in https://imagemagick.org/script/defines.php .

The DNG file may also contain a JPG image that the camera has made from the raw image. 

So you are comparing IM's rendering of the raw image with some other software's rendering of the raw image.

One image can be adjusted to more closely match the colours of another image. One method is by [matching histograms](http://im.snibgo.com/customim.htm#matchhist). Fred has a bash script for this: http://www.fmwconcepts.com/imagemagick/histmatch/index.php . Other methods are available: see [Gain and bias](http://im.snibgo.com/gainbias.htm) and links from the introduction to that page.
